#183568795 optimize choice set edit items

abstracts33d opened the pull request:
![image](https://user-images.githubusercontent.com/16547134/204757342-59ec1279-b9a9-4657-ab13-eeb3d0ebeb4d.png)
![image](https://user-images.githubusercontent.com/16547134/204757504-2bf9f430-8381-4c89-9425-83d0760e4496.png)

skyporter commented on config/initializers/cors.rb:
```diff
@@ -0,0 +1,6 @@
+Rails.application.config.middleware.insert_before 0, Rack::Cors do
+  allow do
+    origins '*'
+    resource '*', headers: :any, methods: [:get, :post, :patch, :put]
+  end
+end
```
pourquoi avoir ajouté cela ?

skyporter replied:
je suis pas sur qu'on peu balancer ca sans explication/argumentation

abstracts33d replied:
rolled back

skyporter commented on app/controllers/react/fields_controller.rb:
```diff
@@ -60,23 +70,23 @@ def choices_for_choice_set
                filter_placeholder: t("catalog_admin.items.reference_editor.reference_editor_filter", locale: params[:locale]),
                loading_message: t("loading", locale: params[:locale]),
-               choices: choices.map { |choice| choice_json_attributes(choice) },
+               choices: filter_category_fields(choices.map{|choice| field.formated_choice(choice)}),
```
```suggestion
               choices: filter_category_fields(choices.map { |choice| field.formated_choice(choice) }),
```

skyporter commented on app/models/field/choice_set.rb:
```diff
@@ -143,15 +143,35 @@ def search_data_as_hash
+      id: choice.id,
+      key: choice.id,
+      label: choice_prefixed_label(choice),
+      has_childrens: choice&.childrens&.any?,
```
```suggestion
      has_childrens: choice.childrens&.any?,
```

skyporter replied:
ceci ne semble pas avoir été changé ?!

skyporter commented on app/models/item_type.rb:
```diff
@@ -38,48 +38,60 @@ class ItemType < ApplicationRecord
-    end
+    return @all_fields if defined? @all_fields
+
+    @all_fields ||= begin
```
a mon avis le begin/end n'est pas nécessaire

abstracts33d replied:
yes

skyporter commented on config/environments/test.rb:
```diff
@@ -54,4 +54,4 @@
 
 # For building URLs in API resource links
-Rails.application.routes.default_url_options = { :host => "localhost:3000" }
+Rails.application.routes.default_url_options = { :host => "localhost:31337" }
```
t'es sur de ton coup la ? pourquoi ce changement ?

abstracts33d replied:
yes j' ai set le port dans la config capybara sinon il change a chaque fois

abstracts33d replied:
rolled back

skyporter commented on config/environments/development.rb:
```diff
@@ -96,6 +96,6 @@
 # For building URLs in API resource links
 Rails.application.routes.default_url_options = {
-  :host => ENV.fetch('DOMAIN', 'localhost:3000'),
+  :host => ENV.fetch('DOMAIN', "localhost:#{Rails::Server::Options.new.parse!(ARGV)[:Port]}"),
```
c'est quoi le soucis avec le fait d'utliser le port 3000 par default ?

abstracts33d replied:
quand tu lance le serveur avec rails s -p xxxx les urls ne fonctionnent plus

abstracts33d replied:
rolled back

skyporter commented on db/migrate/20221130090048_update_editor_component_for_field_choice_set.rb:
```diff
@@ -0,0 +1,5 @@
+class UpdateEditorComponentForFieldChoiceSet < ActiveRecord::Migration[6.1]
+  def change
+    Field::ChoiceSet.update_all(editor_component: 'ChoiceSetInput')
```
Pourquoi ce `update_all` sur les choice set ?

abstracts33d replied:
pour set leur editor_component  non présent avant et qui devient le composant react

skyporter commented on test/integration/api/v1/catalogs_test.rb:
```diff
@@ -22,11 +22,11 @@ class API::V1::CatalogsTest < ActionDispatch::IntegrationTest
     assert_equal(
-      "http://localhost:3000/api/v1/catalogs/#{catalog['slug']}",
+      "http://localhost:31337/api/v1/catalogs/#{catalog['slug']}",
       catalog["_links"]["self"]
     )
     assert_equal(
-      "http://localhost:3000/#{catalog['slug']}",
+      "http://localhost:31337/#{catalog['slug']}",
```
ah nouveau je ne suis pas sur que ce soit une bonne idée de changer le port

abstracts33d replied:
c' est le seul endroit dans les tests ou il est spécifié je peut le laisser sur 3000 et le fixer a 3000 dans capybara mais ca ne changera rien

abstracts33d replied:
rolled back

skyporter commented on test/integration/catalog_admin/choice_sets_test.rb:
```diff
@@ -70,18 +70,19 @@ class CatalogAdmin::ChoiceSetsTest < ActionDispatch::IntegrationTest
+    within(find('#item_one_author_other_languages_uuid_json', visible: false).find(:xpath,".//..")) do
+      find("a[data-target='#choice-modal-#{field.uuid}']", :visible => :all).click
+    end
+    sleep(2)
```
mériterai un commentaire dans le code, car un sleep posé comme ca c'est toujours inquiétant et ce serait bon de savoir pourquoi il est la

abstracts33d replied:
yes

skyporter commented on test/integration/catalog_admin/items_test.rb:
```diff
@@ -25,9 +25,30 @@ class CatalogAdmin::ItemsTest < ActionDispatch::IntegrationTest
+
+    within(find('#item_one_author_language_uuid_json', visible: false).find(:xpath, '..')) do
+      find(".css-g1d714-ValueContainer").click # Click on the filter input
+      sleep(2)
```
meme commentaire que pour le sleep de choice_sets_test.rb

abstracts33d replied:
yes

skyporter commented:
@abstracts33d il faudrait également merger master dans cette branch

skyporter commented on Gemfile.lock:
```diff
@@ -621,6 +623,7 @@ DEPENDENCIES
   pry (~> 0.13)
   puma (~> 5.6)
   pundit (~> 2.1)
+  rack-cors
```
on peut enlever ca non ?

abstracts33d replied:
yes

skyporter commented on config/environments/test.rb:
```diff
@@ -54,4 +54,4 @@
 
 # For building URLs in API resource links
-Rails.application.routes.default_url_options = { :host => "localhost:3000" }
+Rails.application.routes.default_url_options = { :host => "localhost:3000" }
```
tu peux rollback ces changes qui stp?

abstracts33d replied:
yes

abstracts33d committed:
#183568795 - Optimize choice_set edit items

abstracts33d committed:
merge development

daviserez committed:
Fix default values overriding custom values for new item (#485)

* Fix default values overriding custom values for new item

Note: the same problem could happens with "assign_autoincrement_values"
callback, but the case was already taken care with the same logic than
this fix.

Add two tests:
 - One for testing that default values are correctly setted.
 - One for testing that default values don't override custom value.

* Uses guards clauses for better lisibility

* Fix formating and adds comments

daviserez committed:
Hide public list view if field is not human readable (#484)

* Disable server side field option public list view if not human readable

* Hide field option public list view if not human readable

* Disable some options for text field instead of hidding them

* Add help on formatted text field's option

* Migration for setting public list view option according to the human readable status of the field

* Fix no new line at end of file

* Reviews adaptations

 - Adapt help text for better comprehension
 - Add guard clause for better lisibility
 - Use existing bootstrap class instead of inline style

* Simplify formatted text helper

* Simplify formatted text helper (fr)

Co-authored-by: Luca Guindani <luca.guindani@gmail.com>

dependabot[bot] committed:
Bump json5 from 1.0.1 to 1.0.2 (#487)

Bumps [json5](https://github.com/json5/json5) from 1.0.1 to 1.0.2.
- [Release notes](https://github.com/json5/json5/releases)
- [Changelog](https://github.com/json5/json5/blob/main/CHANGELOG.md)
- [Commits](https://github.com/json5/json5/compare/v1.0.1...v1.0.2)

---
updated-dependencies:
- dependency-name: json5
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

daviserez committed:
Fix items custom style (#488)

* Change style description for clarification

* Fix custom style not applied correctly

* Rework headings styles

* Fix dropdown default value not working properly

* Fix and add tests

daviserez committed:
Fix choices presenter (#489)

* Fix the character '>' to be displayed by error for hierarchical choices

* Rubocop and reformat for better lisibility

* Fix choice set presenter test

* Add test for hierarchy choice set presenter

* Rubocop and reformat for better lisibility

* Fix broken choice set test

* Add blank line after guard clause

* Remove empty line

abstracts33d committed:
CR ComplexDatation (#490)

lucaguindani committed:
Localize choice set import file label

lucaguindani committed:
Remove between & outside search conditions for the datation field

lucaguindani committed:
Add datation data to choice_set & choice api responses

lucaguindani committed:
Fix special chars wrongly encoded by strip_tags method
Fix Rubocop correctable issues

lucaguindani committed:
Typo

lucaguindani committed:
Revert docker compose configuration

dependabot[bot] committed:
Bump rack from 2.2.4 to 2.2.6.2 (#492)

Bumps [rack](https://github.com/rack/rack) from 2.2.4 to 2.2.6.2.
- [Release notes](https://github.com/rack/rack/releases)
- [Changelog](https://github.com/rack/rack/blob/main/CHANGELOG.md)
- [Commits](https://github.com/rack/rack/compare/2.2.4...v2.2.6.2)

---
updated-dependencies:
- dependency-name: rack
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump globalid from 1.0.0 to 1.0.1 (#493)

Bumps [globalid](https://github.com/rails/globalid) from 1.0.0 to 1.0.1.
- [Release notes](https://github.com/rails/globalid/releases)
- [Commits](https://github.com/rails/globalid/compare/v1.0.0...v1.0.1)

---
updated-dependencies:
- dependency-name: globalid
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

lucaguindani committed:
Update rails to 6.1.7.1

lucaguindani committed:
Add codeql workflow for github actions

lucaguindani committed:
Fix typo in workflow

lucaguindani committed:
Rename codeql workflow & remove master branch

daviserez committed:
Fix ItemList line view error on homepage (#494)

Fix issue #486

lucaguindani committed:
Bump mammoth js dependency to specific commit

daviserez committed:
Fix export CSV for datation with choice set (#495)

lucaguindani committed:
Hotfix for editor field with deleted user

lucaguindani committed:
Merge branch 'master' into development

abstracts33d committed:
CR

abstracts33d committed:
Retours Complex Datation (#496)

* CR

* pass correct allow_bc_option to AdvancedComplexDatationSearch

* CR

lucaguindani committed:
Fix MultiReferenceEditor search results

lucaguindani committed:
Rollback filtering in MultiReferenceEditor

lucaguindani committed:
Fix pagination for MultiReferenceEditor

daviserez committed:
Fix pictures not displayed in public list display (#497)

* Fix pictures not displayed in public list display

* Restructurates a method for better code homogeneity

abstracts33d committed:
Retours complex datation search 2 (#498)

lucaguindani committed:
Update rubygems to specific version in dockerfiles
Stack versions as envar in dockerfiles

daviserez committed:
Allow to run tests inside docker container (#499)

There are still one error remaining with attachment file.

ERROR CatalogAdmin::ItemsTest#test_import_items_from_CSV (175.05s)
Minitest::UnexpectedError:
ArgumentError: Selenium < 3.14 with remote Chrome doesn't support
multiple file upload
test/integration/catalog_admin/csv_imports_test.rb:10:in `block in
<class:ItemsTest>'

lucaguindani committed:
Update selenium-webdriver to avoid multiple uploads error during tests
Update readme

lucaguindani committed:
Replace infinite scroll in MultiReferenceEditor by a "load more" button (avoid pagination with empty content issue)
Remove filter on fulltext searches
Update tests

daviserez committed:
Fix filterable for summary (#500)

* Fix filterable field not displayed in public list view

* Updates tests

daviserez committed:
Remove unused variable (#501)

daviserez committed:
Updates locales

lucaguindani committed:
Fix de, it locals

daviserez committed:
Tests public list (#502)

* Adds tests for summary view

* Fix broken test

lucaguindani committed:
Fix exclude operator for choice set fields in advanced search
Code style

daviserez committed:
Fix user deletion (#491)

* Rubocop corrections

* Migrations for users.deleted_at

* Fix user deletion

* Unscope user for jti migration

* Use touch when destroying user

* Factorize roles usage

* Fix membership for deleted users

* Adds :inverse_of information and missings has_many relationships for User model

* Unscopes user from searches

And changes dependent params for has_many from nullify to destroy. This
is because searches are used in containers. Deleting a User must not
alter the pages containing a search container.

abstracts33d committed:
CR update ChoiceSetInput

abstracts33d committed:
fix - item summary (#505)

lucaguindani committed:
Fix container with empty item type issue
Add tests

lucaguindani committed:
Fix menu with itself as parent

abstracts33d committed:
merge development

abstracts33d committed:
merge

lucaguindani committed:
Revert docker compose configuration

abstracts33d committed:
CR

abstracts33d committed:
fix category
